test/test_roscpp/test/CMakeLists.txt

```diff
@@ -1,3 +1,5 @@
+SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -pthread")
```

Comment:
As I noted in https://github.com/ros/ros_comm/pull/1014#issuecomment-285128889 this was needed to make the existing tests compile on indigo...